ringpop.go

```diff
@@ -352,6 +359,31 @@ func (rp *Ringpop) setState(s state) {
 	}
 }
 
+// RegisterSelfEvictHook registers the eviction hooks that need to be executed
+// before and after self eviction from the membership. An error will be returned
+// if ringpop was unable to register the hooks. This could happen in the following
+// cases:
+// - Ringpop has not been bootstrapped yet
+// - SelfEvict has already been called
+// - The hook is already registered
+func (rp *Ringpop) RegisterSelfEvictHook(hooks swim.SelfEvictHook) error {
+	if !rp.Ready() {
```

Comment:
The reason to 'prevent' registering hooks is technical. Self-Eviction is part of the SWIM Node which is only initialized after bootstrapping.

I agree that it would be better for usability if it can be registered before bootstrapping, I will see if it can be refactored to accommodate this.